contracts/src/fp/OPSuccinctFaultDisputeGame.sol

```diff
@@ -0,0 +1,499 @@
+        if (proxy.status() != GameStatus.CHALLENGER_WINS) revert InvalidParentGame();
+
+        // Do not allow the game to be initialized if the root claim corresponds to a block at or before the
+        // configured starting block number.
+        if (l2BlockNumber() <= startingOutputRoot.l2BlockNumber) {
+            revert UnexpectedRootClaim(rootClaim());
+        }
+
+        // Set the root claim
+        claimData = ClaimData({
+            parentIndex: parentIndex(),
+            counteredBy: address(0),
+            claimant: gameCreator(),
+            claim: rootClaim(),
+            clock: LibClock.wrap(Duration.wrap(0), Timestamp.wrap(uint64(block.timestamp)))
+        });
+
+        // Hold the bond in the contract.
+        payable(address(this)).transfer(msg.value);
+
+        // Set the game as initialized.
+        initialized = true;
+
+        // Set the game's starting timestamp
+        createdAt = Timestamp.wrap(uint64(block.timestamp));
+    }
+
+    /// @notice The l2BlockNumber of the disputed output root in the `L2OutputOracle`.
+    function l2BlockNumber() public pure returns (uint256 l2BlockNumber_) {
+        l2BlockNumber_ = _getArgUint256(0x54);
+    }
+
+    /// @notice The parent index of the game.
+    function parentIndex() public pure returns (uint32 parentIndex_) {
+        parentIndex_ = _getArgUint32(0x74);
+    }
+
+    /// @notice Only the starting block number of the game.
+    function startingBlockNumber() external view returns (uint256 startingBlockNumber_) {
+        startingBlockNumber_ = startingOutputRoot.l2BlockNumber;
+    }
+
+    /// @notice Starting output root of the game.
+    function startingRootHash() external view returns (Hash startingRootHash_) {
+        startingRootHash_ = startingOutputRoot.root;
+    }
+
+    ////////////////////////////////////////////////////////////////
+    //                    `IDisputeGame` impl                     //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Challenges the game.
+    function challenge() external payable {
+        // INVARIANT: Cannot challenge a game if the game has already been resolved.
+        if (status != GameStatus.IN_PROGRESS) revert ClaimAlreadyResolved();
+
+        // INVARIANT: Cannot challenge a game if the game has already been challenged.
+        if (claimData.counteredBy != address(0)) revert ClaimAlreadyChallenged();
+
+        // INVARIANT: Cannot challenge a game if the clock has already expired.
+        if (getChallengeDuration().raw() == MAX_CHALLENGE_DURATION.raw()) {
+            revert ClockTimeExceeded();
+        }
+
+        // Update the clock to the current block timestamp, which marks the start of the challenge.
+        claimData.clock = LibClock.wrap(Duration.wrap(0), Timestamp.wrap(uint64(block.timestamp)));
+
+        // Hold the bond in the contract.
+        payable(address(this)).transfer(msg.value);
```

Comment:
Should we check that the payable amount matches the expected `challengeBond`?

Reply:
If we want to have two types of bond, one for proposal and one for challenge, we need to fork the factory contract. Need more discussion, so I added a check against init bond, but left a TODO.